packages/addresses/src/program-derived-address.ts

```diff
@@ -0,0 +1,78 @@
+import { assertDigestCapabilityIsAvailable } from '@solana/assertions';
+
+import { Base58EncodedAddress, getBase58EncodedAddressCodec } from './base58';
+import { compressedPointBytesAreOnCurve } from './curve';
+
+type PDAInput = Readonly<{
+    programAddress: Base58EncodedAddress;
+    seeds: Seed[];
+}>;
+type Seed = string | Uint8Array;
```

Comment:
Not suggesting we should keep the existing interface, but do you know how you'd convert from a `Buffer` in the old code to this?

I remember that when I used `Buffer.from([posX, posY])` (posX and posY are 0-255, u8 in rust) as a seed, I couldn't switch it to a `Uint8Array` and get the same result in the old code

I'm not sure if the different implementation here means that `Uint8Array([posX, posY])` would be equivalent?

Reply:
It definitely is. `Buffer` has always been a pain in the ass because it's a Node-only primitive that has required a polyfill in browser, for almost no gain.

https://github.com/solana-labs/solana-web3.js/issues/1100

```ts
> Buffer.from([64, 216])
<Buffer 40 d8>
> new Uint8Array([64, 216])
Uint8Array(2) [ 64, 216 ]
> 0x40 === 64
true
> 0xd8 === 216
true
```

Reply:
> I remember that when I used Buffer.from([posX, posY]) (posX and posY are 0-255, u8 in rust) as a seed, I couldn't switch it to a Uint8Array and get the same result in the old code

Very sus! We should try to repro that and figure out why.